### PR TITLE
MULE-19710: Default jdk xml parsers in AST / DSL api are exposed

### PR DESCRIPTION
**Pull request by fsgonz**

### CODE REVIEW

File: src/main/java/org/apache/xerces/impl/XMLDocumentFragmentScannerImpl.java [hunk 1]
```diff
@@ -0,0 +1,1870 @@
...
+import org.apache.xerces.xni.XMLString;
+import org.apache.xerces.xni.XNIException;
+import org.apache.xerces.xni.parser.XMLComponent;
+import org.apache.xerces.xni.parser.XMLComponentManager;
+import org.apache.xerces.xni.parser.XMLConfigurationException;
+import org.apache.xerces.xni.parser.XMLDocumentScanner;
+import org.apache.xerces.xni.parser.XMLInputSource;
+
+/**
+ * This class was shadowed when fixing MULE-19710. Xerces treats empty CDATA blocks as null. We want to treat it as an empty
```
Comment (elrodro83) [inline src/main/java/org/apache/xerces/impl/XMLDocumentFragmentScannerImpl.java:34]: add a link to the xerces issue as well (https://issues.apache.org/jira/browse/XERCESJ-1033)

File: src/main/java/org/apache/xerces/impl/XMLDocumentFragmentScannerImpl.java [hunk 2]
```diff
@@ -0,0 +1,1870 @@
...
+
+    // call handler
+    if (fDocumentHandler != null) {
+      fDocumentHandler.startCDATA(null);
+    }
+
+    while (true) {
+      fStringBuffer.clear();
+      if (!fEntityScanner.scanData("]]", fStringBuffer)) {
+        // Shadowed in the mule implementation so that the empty
```
Comment (elrodro83) [inline src/main/java/org/apache/xerces/impl/XMLDocumentFragmentScannerImpl.java:1109]: add the issue keys here as well, for easier finding

File: src/main/resources/META-INF/mule-module.properties [hunk 3]
```diff
@@ -4,6 +4,9 @@ artifact.export.classPackages=org.mule.runtime.dsl.api,\
                               org.mule.runtime.dsl.api.component,\
                               org.mule.runtime.dsl.api.component.config,\
                               org.mule.runtime.dsl.api.xml,\
-                              org.mule.runtime.dsl.api.xml.parser
-
+                              org.mule.runtime.dsl.api.xml.parser, \
+                              org.apache.xerces.xni.grammars, \
+                              org.apache.xerces.xni, \
+                              org.apache.xerces.xni.parser, \
+                              org.apache.xerces.xs
```
Comment (elrodro83) [inline src/main/resources/META-INF/mule-module.properties:11]: this might cause trouble on connectos using xerces internally.

**Review by elrodro83 — CHANGES_REQUESTED** [review]

File: api-changes.json [hunk 4]
```diff
@@ -1,4 +1,23 @@
...
+          "old": "parameter org.w3c.dom.Document org.mule.runtime.dsl.api.xml.parser.XmlConfigurationDocumentLoader::loadDocument(java.util.function.Supplier<javax.xml.parsers.SAXParserFactory>, org.xml.sax.EntityResolver, java.lang.String, java.io.InputStream, ===com.sun.org.apache.xerces.internal.xni.grammars.XMLGrammarPool===)",
+          "new": "parameter org.w3c.dom.Document org.mule.runtime.dsl.api.xml.parser.XmlConfigurationDocumentLoader::loadDocument(java.util.function.Supplier<javax.xml.parsers.SAXParserFactory>, org.xml.sax.EntityResolver, java.lang.String, java.io.InputStream, ===org.mule.apache.xerces.xni.grammars.XMLGrammarPool===)",
+          "oldType": "com.sun.org.apache.xerces.internal.xni.grammars.XMLGrammarPool",
+          "newType": "org.mule.apache.xerces.xni.grammars.XMLGrammarPool",
+          "package": "org.mule.runtime.dsl.api.xml.parser",
+          "classSimpleName": "XmlConfigurationDocumentLoader",
+          "methodName": "loadDocument",
+          "parameterIndex": "4",
+          "elementKind": "parameter",
+          "justification": "Use of internal JDK classes"
```
Comment (fsgonz) [inline api-changes.json:16]: This will be removed after merged.

**Review by elrodro83 — APPROVED** [review]

**Pull request merged**